src/pages/Home.js

```diff
@@ -2,37 +2,65 @@ import React, { useEffect, useState } from 'react';
 import ItemList from '../components/ItemList';
 import styled from 'styled-components';
 import ContentWrapper from '../styles/contentWrapper';
-// import mockData from '../db/mock_data.json';
 import axios from 'axios';
+import SearchField from '../components/Search/SearchField';
+import Pagination from '../components/Pagination/Pagination';
+// import mockData from '../db/mock_data.json';
 
 const API_URL =
   'https://gp-super-store-api.herokuapp.com/item/list?size=6&sortDir=asc';
 
 const StyledContainer = styled(ContentWrapper)`
   && {
+    display: flex;
     align-items: center;
-
+    flex-direction: column;
+    // justify-content: space-between;
     width: 100%;
     height: 100%;
-    display: flex;
-    flex-direction: column;
-    justify-content: space-between;
   }
 `;
 /* use this to feed with mockData */
 // const data = { products: mockData.items.slice(0, 6) };
 
 const Home = () => {
   const [productsData, setProductsData] = useState([]);
+  const [totalItems, setTotalItems] = useState(0);
+  const [hasMore, setHasMore] = useState(false);
   const [isFetching, setIsFetching] = useState(false);
+  const [search, setSearch] = useState('');
+  const [fromItem, setFromItem] = useState('');
+  const [page, setPage] = useState(0);
+
+  const onSearch = query => {
+    console.log({ query });
+    onPageChange(0);
+    setSearch(`&q=${query}`);
+  };
+
+  const onPageChange = pageNum => {
+    console.log({ pageNum });
+    // check if already in last page
+    if (pageNum * 6 >= totalItems && !hasMore) return;
+    setPage(pageNum);
+    setFromItem(`&from=${pageNum * 6}`);
+  };
+
   useEffect(() => {
+    const queryParams = search + fromItem;
     const fetchData = async () => {
       try {
         setIsFetching(true);
-
-        const response = await axios.get(API_URL);
+        let response;
+        if (queryParams) {
+          response = await axios.get(API_URL + queryParams);
```

Comment:
You can improve the way you use axios by utilizing the "options" argument on `axios.get`. You can use options to pass in query params so you don't have to append them to your url. It makes things look a lot nicer, and it's easier to use.

Reply:
Since you use this axios query in multiple files (Home and Deals) you can move this into a `/src/utils/api.js` file and import it into your Home and Deals files.

Reply:
copied!